OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,407 @@
+function getMerkleProof(txHash: string): Promise<MerkleProof>
+function getStorage(contract: string, key: string): Promise<string>
+function getAllowance(asset: Asset, fromAddress: string, toAddress: string): Promise<number>
+function getBlock(block: number | string): Promise<Block>
+function getTransaction(txHash: string): Promise<Transaction>
+function getNetwork(): Network
+function getBalance(address: string): Promise<Balance>
+</pre>
+
+For further explanation about the wrapped method consult https://ontio.github.io/documentation/restful_api_en.html . The types '''Transaction''', '''Block''', '''MerkleProof''' and '''Balance''' corresponds to the exact object returned from Ontology blockchain.
+
+TODO: copy the definition of all network api calls with inputs and outputs from ontology documentation.
+
+===Provider===
+A primary focus of Provider API is to check if user has installed any dAPI provider and to get information about it.
+
+====getProvider====
+<pre>
+
+interface Provider {
+  name: string;
+  version: string;
+  compatibility: string[];
+}
+
+function getProvider(): Promise<Provider>
+</pre>
+
+Returns the information about installed dAPI provider if there is one installed. <code>compatibility</code> refers to all OEPs this dAPI provider supports. E.g.:
+
+<pre>
+compatability: [
+    'OEP-6',
+    'OEP-10',
+    'OEP-29'
+]
+</pre>
+
+dAPI provider does not need to support all future OEPs concerned with dAPI.
+
+* Rejects with '''NO_PROVIDER''' in case there is no dAPI provider installed.
+
+===Asset===
+A primary focus of Asset API is to initiate a transfer. The request needs to be confirmed by the user.
+
+====getDefaultAccount====	
+<pre>	
+function getDefaultAccount(): Promise<string>	
+</pre>	
+Returns currently selected account of logged in user. 	
+ * Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+
+====makeTransfer====
+<pre>
+function makeTransfer(recipient: string, asset: Asset, amount: number): Promise<string>
+</pre>
+
+Initiates a transfer of <code>amount asset</code> to <code>recipient</code> account. 
+
+The amount should always be an integer value. Therefore is is represented in the smallest, further non divisible units for the asset. E.g.: If the <code>asset</code> is ONG then the value is multiplied by 1000000000.
+
+Returns transaction Id.
+
+* Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+* Rejects with '''MALFORMED_ACCOUNT''' in case the <code>recipient</code> is not a valid account
+* Rejects with '''CANCELED''' in case the user cancels the request
+
+===Identity===
+A primary focus of Identity API is to interact with ONT ID smart contract on the blockchain.
+
+====getDefaultIdentity====	
+<pre>	
+function getDefaultIdentity(): Promise<string>	
+</pre>	
+ Returns currently selected identity of logged in user.	
+ * Rejects with '''NO_IDENTITY''' in case the user is not signed in or has no identity
+
+====getPublicKeys====
+<pre>
```

Comment:
Is this method required, as I believe the the public keys are a part of the DDO returned by `getDDO`.

Reply:
It's just that, those methods are separate on the ONT ID SC. But you are right, that DDO will also contain the public keys. In time, there might be many attributes (the other thing in the ddo) which will slow down the query, when you just want to get the public keys (e.g: message verification?)

Reply:
It seems that the DDO can be retrieved in a single query:
https://ontio.github.io/documentation/ontology_ts_sdk_identity_claim_en.html#13-query-ddo